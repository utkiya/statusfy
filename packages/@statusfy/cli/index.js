// Inspired on https://github.com/vuejs/vuepress/blob/master/packages/%40vuepress/cli/index.js
const program = require('commander')
const satisfies = require('semver/functions/satisfies')
const { chalk, path } = require('@statusfy/common')

const packageError = (name) => {
  console.log(chalk.red(
    `\n[statusfy] @statusfy/cli ` +
    `requires ${name} to be installed.\n` +
    `${err}`
  ))
}

try {
  require.resolve('@statusfy/core')
} catch (err) {
  packageError('@statusfy/core')
  process.exit(1)
}

try {
  require.resolve('@statusfy/editor')
} catch (err) {
  packageError('@statusfy/editor')
  process.exit(1)
}

const pkg = require('@statusfy/core/package.json')
const requiredVersion = pkg.engines.node

if (!satisfies(process.version, requiredVersion)) {
  console.log(chalk.red(
    `Minimum Node version not met:\n` +
    `You are using Node ${process.version}, ` +
    `but Statusfy requires Node ${requiredVersion}.`
  ))
  console.log(chalk.red(`Please upgrade your Node version.`))

  process.exit(1)
}

const {
  init,
  dev,
  build,
  generate,
  start,
  newIncident,
  deleteIncident,
  updateIncident
} = require('@statusfy/core/lib')
const sourceDir = path.resolve('.')

program
  .version(pkg.version)
  .usage('<command> [options]')

program
  .command('init')
  .description('Create a base project')
  .option('-d, --dir <outDir>', 'specify the installation directory')
  .action(({ dir }) => {
    const outDir = dir ? path.resolve(dir) : null
    wrapCommand(init)(sourceDir, { outDir })
  })

program
  .command('dev')
  .description('Starts the application in development mode (hot-code reloading, error reporting, etc.).')
  .option('-p, --port <port>', 'use specified port (default: 3000)')
  .option('-s, --ssr', 'force SSR')
  .action(({ port, ssr }) => {
    wrapCommand(dev)(sourceDir, { port, ssr })
  })

program
  .command('build')
  .description('Compiles the application for production deployment')
  .option('-a, --analyze', 'launch the final bundle analysis')
  .action(({ analyze }) => {
    wrapCommand(build)(sourceDir, { analyze })
  })

program
  .command('generate')
  .description('Generate a static web application (server-rendered)')
  .option('-d, --dest <outDir>', 'specify generate output dir (default: ./dist)')
  .option('-a, --analyze', 'launch the final bundle analysis')
  .action(({ dest, analyze }) => {
    const outDir = dest ? path.resolve(dest) : null
    wrapCommand(generate)(sourceDir, { outDir, analyze })
  })

program
  .command('start')
  .description(`Starts the application in production mode. The application should be compiled with ${chalk.cyan(`statusfy build`)} first.`)
  .option('-p, --port <port>', 'use specified port (default: 3000)')
  .option('-H, --host <host>', 'use specified host (default: 127.0.0.1)')
  .action(({ host, port }) => {
    wrapCommand(start)(sourceDir, { host, port })
  })

program
  .command('new-incident')
  .description(`Creates a new incident after answering a few questions.`)
  .action(() => {
    wrapCommand(newIncident)(sourceDir, { })
  })

program
  .command('delete-incident')
  .description(`Delete an incident including translations.`)
  .action(() => {
    wrapCommand(deleteIncident)(sourceDir, { })
  })


program
  .command('update-incident')
  .description(`Update an incident including translations.`)
  .action(() => {
    wrapCommand(updateIncident)(sourceDir, {})
  })

// output help information on unknown commands
program
  .arguments('<command>')
  .action((cmd) => {
    program.outputHelp()
    console.log(`  ` + chalk.red(`Unknown command ${chalk.yellow(cmd)}.\n`))
  })

// add some useful info on help
program.on('--help', () => {
  console.log(`\n  Run ${chalk.cyan(`statusfy <command> --help`)} for detailed usage of given command.\n`)
})

program.commands.forEach(c => c.on('--help', () => console.log()))
program.parse(process.argv)

if (!process.argv.slice(2).length) {
  program.outputHelp()
}

function wrapCommand (fn) {
  return (...args) => {
    return fn(...args).catch(err => {
      console.error(chalk.red(err.stack))
      process.exitCode = 1
    })
  }
}
