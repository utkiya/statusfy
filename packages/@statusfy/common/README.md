# @statusfy/common

> Common Utils for Statusfy
