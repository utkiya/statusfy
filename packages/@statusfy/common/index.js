const path = require('upath')
const esm = require('./lib/esm')

exports.logger = require('./lib/logger')
exports.validator = require('./lib/validator')
exports.grayMatter = require('./lib/gray-matter')
exports.slugify = require('./lib/slugify')
exports.generateDemoContent = require('./lib/generate-content')
exports.style = require('./lib/style')
exports.postcss = require('./lib/postcss')
exports.esm = esm
exports.Dates = esm(path.join(__dirname, './lib/dates.js')).default

exports.url = require('url')
exports.toml = require('toml')
exports.tomlify = require('tomlify-j0.4')
exports.yaml = require('yaml')

exports.chalk = require('chalk')
exports.fse = require('fs-extra')
exports.hash = require('hash-sum')
exports.path = path
exports.LRU = require('lru-cache')
