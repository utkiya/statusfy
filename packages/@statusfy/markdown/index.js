const emoji = require('markdown-it-emoji')

const containers = require('./containers')
const frontmatter = require('./frontmatter')
const links = require('./links')

module.exports = ({ markdown = {} } = {}) => {
  const md = require('markdown-it')({
    html: true
  })
  // custom plugins
  .use(containers)
  .use(frontmatter)
  .use(links)

  // 3rd party plugins
  .use(emoji)

  // apply user config
  if (markdown.config) {
    markdown.config(md)
  }

  return md
}
