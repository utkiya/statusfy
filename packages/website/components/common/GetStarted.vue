<template>
  <section class="bg-white py-8">
    <div class="container my-8">
      <div class="flex lg:items-center flex-col lg:flex-row lg:justify-between">
        <div class="mb-8 lg:mb-0 lg:pr-4">
          <h2 class="text-2xl font-normal leading-none mb-3">
            {{ $t('getStarted.title') }}
          </h2>
          <h3 class="text-xl text-grey-dark font-normal leading-tight">
            {{ $t('getStarted.description') }}
          </h3>
        </div>
        <div class="lg:pl-4">
          <div class="flex items-center">
            <a
              :href="$t('links.getStarted.url')"
              class="btn btn-blue mr-4"
              target="__blank"
              rel="noopener"
            >
              {{ $t('links.getStarted.title') }}
            </a>
            <a
              :href="$t('links.github.url')"
              class="btn btn-grey"
              target="__blank"
              rel="noopener"
            >
              {{ $t('links.github.title') }}
            </a>
          </div>
        </div>
      </div>
    </div>
  </section>
</template>
