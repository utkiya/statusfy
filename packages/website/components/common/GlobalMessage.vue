<template>
  <div
    v-if="$t('globalMessage') !== ''"
    class="bg-orange text-white text-center py-2 px-4 h-16"
  >
    <div class="p-2 items-center leading-none flex inline-flex h-full leading-tight">
      <!-- eslint-disable vue/no-v-html -->
      <div
        class="font-semibold mr-2 flex-auto"
        v-html="$t('globalMessage')"
      />
      <!-- eslint-enable -->
    </div>
  </div>
</template>

<style scoped>
div >>> a {
  @apply text-grey-lightest underline;

  &:hover {
    @apply text-grey-lighter;
  }
}
</style>
