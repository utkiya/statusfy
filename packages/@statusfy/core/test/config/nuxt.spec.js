import isString from "lodash.isstring";
import cloneDeepWith from "lodash.clonedeepwith";
import { path } from "@statusfy/common";

import generateConfig from "@/core/lib/config/generate";
const tempPath = path.resolve(__dirname, "../.tmp");

const relative = dir => {
  return path.relative(__dirname, dir);
};

describe("config:nuxt", () => {
  const executeConfigTest = label => {
    const sourceDir = path.resolve(tempPath, label);

    const generatedConfig = generateConfig(sourceDir, {});
    /**
     * @type { import("@nuxt/types").Configuration }
     */
    let nuxtConfig = generatedConfig.nuxtConfig;

    // Check absolute dirs
    expect(path.isAbsolute(nuxtConfig.buildDir)).toBeTruthy();
    expect(path.isAbsolute(nuxtConfig.generate.dir)).toBeTruthy();
    expect(path.isAbsolute(nuxtConfig.icon.iconSrc)).toBeTruthy();
    expect(path.isAbsolute(nuxtConfig.rootDir)).toBeTruthy();
    // expect(path.isAbsolute(nuxtConfig.srcDir)).toBeTruthy()
    expect(path.isAbsolute(nuxtConfig.statusfy.assets.mainLogo)).toBeTruthy();

    if (nuxtConfig.statusfy.publicFilesPath) {
      expect(path.isAbsolute(nuxtConfig.statusfy.publicFilesPath)).toBeTruthy();
    }
    expect(
      path.isAbsolute(nuxtConfig.statusfy.siteConfig.sourceDir)
    ).toBeTruthy();

    if (nuxtConfig.workbox) {
      expect(path.isAbsolute(nuxtConfig.workbox.globDirectory)).toBeTruthy();
    }

    // eslint-disable-next-line no-unused-vars
    for (const dir of nuxtConfig.modulesDir) {
      expect(isString(dir)).toBeTruthy();
    }

    expect(nuxtConfig.modulesDir.length).toBeGreaterThanOrEqual(2);

    // Convert absolute paths to relative
    nuxtConfig = cloneDeepWith(nuxtConfig, value => {
      if (isString(value) && path.isAbsolute(value)) {
        const relativePath = path.normalize(relative(value));

        if (relativePath.split("../").length <= 3) {
          return path.toUnix(path.normalize(relative(value)));
        }
      }
    });

    // Check config
    expect(nuxtConfig).toMatchSnapshot();
  };

  test("simple", () => {
    executeConfigTest("simple");
  });

  test("cli", () => {
    executeConfigTest("cli");
  });

  test("advanced", () => {
    executeConfigTest("advanced");
  });

  test("spanish", () => {
    executeConfigTest("spanish");
  });
});
