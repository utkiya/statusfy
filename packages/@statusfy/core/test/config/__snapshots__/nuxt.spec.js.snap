// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`config:nuxt advanced 1`] = `
Object {
  "axios": Object {
    "browserBaseURL": "/",
  },
  "babel": Object {
    "plugins": Array [
      "@babel/transform-runtime",
    ],
  },
  "build": Object {
    "analyze": true,
    "babel": Object {
      "presets": [Function],
    },
    "extend": [Function],
    "postcss": Object {
      "plugins": Object {
        "autoprefixer": Object {},
        "postcss-combine-duplicated-selectors": Object {},
        "postcss-import": Object {},
        "postcss-nested": Object {},
        "postcss-preset-env": Object {},
        "tailwindcss": "../../tailwind.js",
      },
    },
    "publicPath": "/static/",
    "router": Object {
      "linkActiveClass": "active",
      "linkExactActiveClass": "exact",
      "scrollBehavior": [Function],
    },
  },
  "buildDir": "../.tmp/advanced/.statusfy",
  "css": Array [
    "~assets/css/tailwind.css",
    "@fortawesome/fontawesome-svg-core/styles.css",
    "../.tmp/advanced/theme/default/style.css",
    "../.tmp/advanced/theme/default/style.less",
    "../.tmp/advanced/theme/default/style.sass",
    "../.tmp/advanced/theme/default/style.scss",
    "../.tmp/advanced/theme/default/style.styl",
    "../.tmp/advanced/theme/default/style.stylus",
  ],
  "dev": true,
  "env": Object {
    "isDev": true,
  },
  "generate": Object {
    "dir": "../.tmp/advanced/dist",
  },
  "head": Object {
    "link": Array [
      Object {
        "color": "#3e4e88",
        "href": "/icons/safari-pinned-tab.svg",
        "rel": "mask-icon",
      },
    ],
    "meta": Array [
      Object {
        "charset": "utf-8",
      },
      Object {
        "content": "width=device-width, initial-scale=1",
        "name": "viewport",
      },
      Object {
        "content": "Statusfy",
        "name": "generator",
      },
      Object {
        "content": "Description",
        "hid": "description",
        "name": "description",
      },
    ],
    "title": "Long Title",
    "titleTemplate": "%s | Long Title",
  },
  "icon": Object {
    "iconSrc": "../.tmp/advanced/theme/default/img/icon.png",
    "sizes": Array [
      16,
      120,
      144,
      152,
      192,
      384,
      512,
    ],
  },
  "loading": Object {
    "color": "#1b1f23",
  },
  "manifest": Object {
    "description": "Description",
    "lang": "en",
    "name": "Long Title",
    "publicPath": "/static/",
    "short_name": "Short Title",
    "start_url": "/?standalone=true&utm_source=web_app&utm_medium=pwa",
    "version": "0.5.0",
  },
  "meta": Object {
    "author": "Statusfy",
    "charset": "utf-8",
    "mobileApp": false,
    "mobileAppIOS": false,
    "ogDescription": false,
    "ogHost": "https://demo.statusfy.co",
    "ogTitle": false,
    "theme_color": "#1b1f23",
  },
  "mode": "universal",
  "modules": Array [
    "@nuxtjs/onesignal",
    "~/modules/statusfy",
    "@nuxtjs/axios",
    Array [
      "nuxt-i18n",
      Object {
        "baseUrl": "https://demo.statusfy.co",
        "defaultLocale": "en",
        "detectBrowserLanguage": Object {
          "cookieKey": "app_name.lang_redirected",
          "useCookie": true,
        },
        "langDir": "locales/",
        "lazy": true,
        "locales": Array [
          Object {
            "code": "en",
            "file": "en.js",
            "iso": "en-US",
            "name": "English",
          },
          Object {
            "code": "es",
            "file": "es.js",
            "iso": "es-ES",
            "name": "Español",
          },
        ],
        "seo": false,
        "vueI18n": Object {
          "fallbackLocale": "en",
          "silentTranslationWarn": false,
        },
      },
    ],
    Array [
      "@nuxtjs/pwa",
      Object {
        "manifest": false,
        "workbox": false,
      },
    ],
  ],
  "modulesDir": Array [
    "../../node_modules",
    "../../../node_modules",
    "../.tmp/simple/node_modules",
    "../.tmp/cli/node_modules",
    "../.tmp/advanced/node_modules",
  ],
  "oneSignal": Object {
    "init": Object {
      "allowLocalhostAsSecureOrigin": true,
      "appId": "onesignal-app-id",
      "welcomeNotification": Object {
        "disable": true,
      },
    },
  },
  "plugins": Array [
    "~/plugins/axios",
    "~/plugins/vue-svgicon",
  ],
  "rootDir": "../..",
  "server": Object {
    "host": "127.0.0.1",
    "port": 3000,
  },
  "srcDir": "../../client",
  "statusfy": Object {
    "assets": Object {
      "mainLogo": "../.tmp/advanced/theme/default/img/logo.svg",
    },
    "dev": true,
    "iconSizes": Array [
      16,
      120,
      144,
      152,
      192,
      384,
      512,
    ],
    "langDir": "locales/",
    "locales": Array [
      Object {
        "code": "en",
        "iso": "en-US",
        "name": "English",
      },
      Object {
        "code": "es",
        "iso": "es-ES",
        "name": "Español",
      },
    ],
    "publicFilesPath": "../.tmp/advanced/public",
    "siteConfig": Object {
      "analytics": Object {
        "ga": "UA-XXXXXXXXX-Y",
      },
      "baseUrl": "https://demo.statusfy.co",
      "build": Object {
        "isStatic": false,
      },
      "content": Object {
        "dir": "content",
        "frontMatterFormat": "yaml",
        "systems": Array [
          "cdn",
          "conversions",
          "site-delivery",
          "api",
        ],
      },
      "defaultLocale": "en",
      "description": "Description",
      "head": Object {
        "link": Array [
          Object {
            "color": "#3e4e88",
            "href": "/icons/safari-pinned-tab.svg",
            "rel": "mask-icon",
          },
        ],
        "meta": Array [
          Object {
            "charset": "utf-8",
          },
          Object {
            "content": "width=device-width, initial-scale=1",
            "name": "viewport",
          },
          Object {
            "content": "Statusfy",
            "name": "generator",
          },
          Object {
            "content": "Description",
            "hid": "description",
            "name": "description",
          },
        ],
        "title": "Long Title",
        "titleTemplate": "%s | Long Title",
      },
      "locales": Array [
        Object {
          "code": "en",
          "iso": "en-US",
          "name": "English",
        },
        Object {
          "code": "es",
          "iso": "es-ES",
          "name": "Español",
        },
      ],
      "manifest": true,
      "name": "app_name",
      "notifications": Object {
        "feeds": true,
        "icalendar": true,
        "support": false,
        "twitter": false,
        "webpush": Object {
          "onesignal": Object {
            "appId": "onesignal-app-id",
          },
        },
      },
      "serviceWorker": true,
      "short_title": "Short Title",
      "sourceDir": "../.tmp/advanced",
      "theme": Object {
        "scheduled": Object {
          "position": "belowSystems",
        },
      },
      "title": "Long Title",
    },
    "sourceDir": "../.tmp/advanced",
    "version": "0.5.0",
  },
  "workbox": Object {
    "assetsURLPattern": "/static/(?!content).*$",
    "cacheId": "app_name",
    "cleanupOutdatedCaches": true,
    "globDirectory": "../.tmp/advanced/.statusfy/dist/client",
    "offlineAnalytics": true,
    "publicPath": "/static/",
    "runtimeCaching": Array [
      Object {
        "strategyOptions": Object {
          "cacheExpiration": Object {
            "maxAgeSeconds": 300,
            "maxEntries": 10,
          },
          "cacheName": "app_name_title_api",
        },
        "urlPattern": "(/static/content)?/api/v0/.*",
      },
    ],
  },
}
`;

exports[`config:nuxt cli 1`] = `
Object {
  "axios": Object {
    "browserBaseURL": "/",
  },
  "babel": Object {
    "plugins": Array [
      "@babel/transform-runtime",
    ],
  },
  "build": Object {
    "analyze": true,
    "babel": Object {
      "presets": [Function],
    },
    "extend": [Function],
    "postcss": Object {
      "plugins": Object {
        "autoprefixer": Object {},
        "postcss-combine-duplicated-selectors": Object {},
        "postcss-import": Object {},
        "postcss-nested": Object {},
        "postcss-preset-env": Object {},
        "tailwindcss": "../../tailwind.js",
      },
    },
    "publicPath": "/static/",
    "router": Object {
      "linkActiveClass": "active",
      "linkExactActiveClass": "exact",
      "scrollBehavior": [Function],
    },
  },
  "buildDir": "../.tmp/cli/.statusfy",
  "css": Array [
    "~assets/css/tailwind.css",
    "@fortawesome/fontawesome-svg-core/styles.css",
  ],
  "dev": true,
  "env": Object {
    "isDev": true,
  },
  "generate": Object {
    "dir": "../.tmp/cli/dist",
  },
  "head": Object {
    "meta": Array [
      Object {
        "charset": "utf-8",
      },
      Object {
        "content": "width=device-width, initial-scale=1",
        "name": "viewport",
      },
      Object {
        "content": "Statusfy",
        "name": "generator",
      },
      Object {
        "content": "",
        "hid": "description",
        "name": "description",
      },
    ],
    "title": "Title",
    "titleTemplate": "%s | Title",
  },
  "icon": Object {
    "iconSrc": "../../client/static/icon.png",
    "sizes": Array [
      16,
      120,
      144,
      152,
      192,
      384,
      512,
    ],
  },
  "loading": Object {
    "color": "#1b1f23",
  },
  "manifest": Object {
    "description": "",
    "lang": "en",
    "name": "Title",
    "publicPath": "/static/",
    "short_name": "Title",
    "start_url": "/?standalone=true&utm_source=web_app&utm_medium=pwa",
    "version": "0.5.0",
  },
  "meta": Object {
    "author": "Statusfy",
    "charset": "utf-8",
    "mobileApp": false,
    "mobileAppIOS": false,
    "ogDescription": false,
    "ogHost": "/",
    "ogTitle": false,
    "theme_color": "#1b1f23",
  },
  "mode": "universal",
  "modules": Array [
    "~/modules/statusfy",
    "@nuxtjs/axios",
    Array [
      "nuxt-i18n",
      Object {
        "baseUrl": null,
        "defaultLocale": "en",
        "detectBrowserLanguage": Object {
          "cookieKey": "title.lang_redirected",
          "useCookie": true,
        },
        "langDir": "locales/",
        "lazy": true,
        "locales": Array [
          Object {
            "code": "en",
            "file": "en.js",
            "iso": "en-US",
            "name": "English",
          },
        ],
        "seo": false,
        "vueI18n": Object {
          "fallbackLocale": "en",
          "silentTranslationWarn": false,
        },
      },
    ],
    Array [
      "@nuxtjs/pwa",
      Object {
        "manifest": false,
        "workbox": false,
      },
    ],
  ],
  "modulesDir": Array [
    "../../node_modules",
    "../../../node_modules",
    "../.tmp/simple/node_modules",
    "../.tmp/cli/node_modules",
  ],
  "plugins": Array [
    "~/plugins/axios",
    "~/plugins/vue-svgicon",
  ],
  "rootDir": "../..",
  "server": Object {
    "host": "127.0.0.1",
    "port": 3000,
  },
  "srcDir": "../../client",
  "statusfy": Object {
    "assets": Object {
      "mainLogo": "../../client/assets/img/logo.svg",
    },
    "dev": true,
    "iconSizes": Array [
      16,
      120,
      144,
      152,
      192,
      384,
      512,
    ],
    "langDir": "locales/",
    "locales": Array [
      Object {
        "code": "en",
        "iso": "en-US",
        "name": "English",
      },
    ],
    "publicFilesPath": "../.tmp/cli/public",
    "siteConfig": Object {
      "baseUrl": "/",
      "build": Object {
        "isStatic": false,
      },
      "content": Object {
        "dir": "content",
        "frontMatterFormat": "yaml",
        "systems": Array [
          "api",
          "cdn",
          "dns",
          "site-delivery",
        ],
      },
      "defaultLocale": "en",
      "description": "",
      "locales": Array [
        Object {
          "code": "en",
          "iso": "en-US",
          "name": "English",
        },
      ],
      "manifest": true,
      "name": "title",
      "notifications": Object {
        "feeds": true,
        "icalendar": true,
        "support": false,
        "twitter": false,
      },
      "serviceWorker": true,
      "short_title": "Title",
      "sourceDir": "../.tmp/cli",
      "theme": Object {
        "scheduled": Object {
          "position": "belowSystems",
        },
      },
      "title": "Title",
    },
    "sourceDir": "../.tmp/cli",
    "version": "0.5.0",
  },
  "workbox": Object {
    "assetsURLPattern": "/static/(?!content).*$",
    "cacheId": "title",
    "cleanupOutdatedCaches": true,
    "globDirectory": "../.tmp/cli/.statusfy/dist/client",
    "publicPath": "/static/",
    "runtimeCaching": Array [
      Object {
        "strategyOptions": Object {
          "cacheExpiration": Object {
            "maxAgeSeconds": 300,
            "maxEntries": 10,
          },
          "cacheName": "title_api",
        },
        "urlPattern": "(/static/content)?/api/v0/.*",
      },
    ],
  },
}
`;

exports[`config:nuxt simple 1`] = `
Object {
  "axios": Object {
    "browserBaseURL": "/",
  },
  "babel": Object {
    "plugins": Array [
      "@babel/transform-runtime",
    ],
  },
  "build": Object {
    "analyze": true,
    "babel": Object {
      "presets": [Function],
    },
    "extend": [Function],
    "postcss": Object {
      "plugins": Object {
        "autoprefixer": Object {},
        "postcss-combine-duplicated-selectors": Object {},
        "postcss-import": Object {},
        "postcss-nested": Object {},
        "postcss-preset-env": Object {},
        "tailwindcss": "../../tailwind.js",
      },
    },
    "publicPath": "/static/",
    "router": Object {
      "linkActiveClass": "active",
      "linkExactActiveClass": "exact",
      "scrollBehavior": [Function],
    },
  },
  "buildDir": "../.tmp/simple/.statusfy",
  "css": Array [
    "~assets/css/tailwind.css",
    "@fortawesome/fontawesome-svg-core/styles.css",
  ],
  "dev": true,
  "env": Object {
    "isDev": true,
  },
  "generate": Object {
    "dir": "../.tmp/simple/dist",
  },
  "head": Object {
    "meta": Array [
      Object {
        "charset": "utf-8",
      },
      Object {
        "content": "width=device-width, initial-scale=1",
        "name": "viewport",
      },
      Object {
        "content": "Statusfy",
        "name": "generator",
      },
      Object {
        "content": "",
        "hid": "description",
        "name": "description",
      },
    ],
    "title": "Title",
    "titleTemplate": "%s | Title",
  },
  "icon": Object {
    "iconSrc": "../../client/static/icon.png",
    "sizes": Array [
      16,
      120,
      144,
      152,
      192,
      384,
      512,
    ],
  },
  "loading": Object {
    "color": "#1b1f23",
  },
  "meta": Object {
    "author": "Statusfy",
    "charset": "utf-8",
    "mobileApp": false,
    "mobileAppIOS": false,
    "ogDescription": false,
    "ogHost": "https://simple.example.com",
    "ogTitle": false,
    "theme_color": "#1b1f23",
  },
  "mode": "universal",
  "modules": Array [
    "~/modules/statusfy",
    "@nuxtjs/axios",
    Array [
      "nuxt-i18n",
      Object {
        "baseUrl": "https://simple.example.com",
        "defaultLocale": "en",
        "detectBrowserLanguage": Object {
          "cookieKey": "title.lang_redirected",
          "useCookie": true,
        },
        "langDir": "locales/",
        "lazy": true,
        "locales": Array [
          Object {
            "code": "en",
            "file": "en.js",
            "iso": "en-US",
            "name": "English",
          },
        ],
        "seo": false,
        "vueI18n": Object {
          "fallbackLocale": "en",
          "silentTranslationWarn": false,
        },
      },
    ],
    Array [
      "@nuxtjs/pwa",
      Object {
        "manifest": false,
        "workbox": false,
      },
    ],
  ],
  "modulesDir": Array [
    "../../node_modules",
    "../../../node_modules",
    "../.tmp/simple/node_modules",
  ],
  "plugins": Array [
    "~/plugins/axios",
    "~/plugins/vue-svgicon",
  ],
  "rootDir": "../..",
  "server": Object {
    "host": "127.0.0.1",
    "port": 3000,
  },
  "srcDir": "../../client",
  "statusfy": Object {
    "assets": Object {
      "mainLogo": "../../client/assets/img/logo.svg",
    },
    "dev": true,
    "iconSizes": Array [
      16,
      120,
      144,
      152,
      192,
      384,
      512,
    ],
    "langDir": "locales/",
    "locales": Array [
      Object {
        "code": "en",
        "iso": "en-US",
        "name": "English",
      },
    ],
    "publicFilesPath": null,
    "siteConfig": Object {
      "baseUrl": "https://simple.example.com",
      "build": Object {
        "isStatic": false,
      },
      "content": Object {
        "dir": "content",
        "frontMatterFormat": "yaml",
        "systems": Array [
          "api",
          "cdn",
          "dns",
          "site-delivery",
        ],
      },
      "defaultLocale": "en",
      "description": "",
      "locales": Array [
        Object {
          "code": "en",
          "iso": "en-US",
          "name": "English",
        },
      ],
      "manifest": false,
      "name": "title",
      "notifications": false,
      "serviceWorker": false,
      "short_title": "Title",
      "sourceDir": "../.tmp/simple",
      "theme": Object {
        "scheduled": Object {
          "position": "belowSystems",
        },
      },
      "title": "Title",
    },
    "sourceDir": "../.tmp/simple",
    "version": "0.5.0",
  },
}
`;

exports[`config:nuxt spanish 1`] = `
Object {
  "axios": Object {
    "browserBaseURL": "/",
  },
  "babel": Object {
    "plugins": Array [
      "@babel/transform-runtime",
    ],
  },
  "build": Object {
    "analyze": true,
    "babel": Object {
      "presets": [Function],
    },
    "extend": [Function],
    "postcss": Object {
      "plugins": Object {
        "autoprefixer": Object {},
        "postcss-combine-duplicated-selectors": Object {},
        "postcss-import": Object {},
        "postcss-nested": Object {},
        "postcss-preset-env": Object {},
        "tailwindcss": "../../tailwind.js",
      },
    },
    "publicPath": "/static/",
    "router": Object {
      "linkActiveClass": "active",
      "linkExactActiveClass": "exact",
      "scrollBehavior": [Function],
    },
  },
  "buildDir": "../.tmp/spanish/.statusfy",
  "css": Array [
    "~assets/css/tailwind.css",
    "@fortawesome/fontawesome-svg-core/styles.css",
    "../.tmp/advanced/theme/default/style.css",
    "../.tmp/advanced/theme/default/style.less",
    "../.tmp/advanced/theme/default/style.sass",
    "../.tmp/advanced/theme/default/style.scss",
    "../.tmp/advanced/theme/default/style.styl",
    "../.tmp/advanced/theme/default/style.stylus",
  ],
  "dev": true,
  "env": Object {
    "isDev": true,
  },
  "generate": Object {
    "dir": "../.tmp/spanish/dist",
  },
  "head": Object {
    "meta": Array [
      Object {
        "charset": "utf-8",
      },
      Object {
        "content": "width=device-width, initial-scale=1",
        "name": "viewport",
      },
      Object {
        "content": "Statusfy",
        "name": "generator",
      },
      Object {
        "content": "Descripción",
        "hid": "description",
        "name": "description",
      },
    ],
    "title": "Título",
    "titleTemplate": "%s | Título",
  },
  "icon": Object {
    "iconSrc": "../.tmp/advanced/theme/default/img/icon.png",
    "sizes": Array [
      16,
      120,
      144,
      152,
      192,
      384,
      512,
    ],
  },
  "loading": Object {
    "color": "#1b1f23",
  },
  "manifest": Object {
    "description": "Descripción",
    "lang": "es",
    "name": "Título",
    "publicPath": "/static/",
    "short_name": "Título",
    "start_url": "/?standalone=true&utm_source=web_app&utm_medium=pwa",
    "version": "0.5.0",
  },
  "meta": Object {
    "author": "Statusfy",
    "charset": "utf-8",
    "mobileApp": false,
    "mobileAppIOS": false,
    "ogDescription": false,
    "ogHost": "https://demo-es.statusfy.co",
    "ogTitle": false,
    "theme_color": "#1b1f23",
  },
  "mode": "universal",
  "modules": Array [
    "@nuxtjs/onesignal",
    "~/modules/statusfy",
    "@nuxtjs/axios",
    Array [
      "nuxt-i18n",
      Object {
        "baseUrl": "https://demo-es.statusfy.co",
        "defaultLocale": "es",
        "detectBrowserLanguage": Object {
          "cookieKey": "Nombre Aplicación.lang_redirected",
          "useCookie": true,
        },
        "langDir": "locales/",
        "lazy": true,
        "locales": Array [
          Object {
            "code": "es",
            "file": "es.js",
            "iso": "es-ES",
            "name": "Español",
          },
          Object {
            "code": "fr",
            "file": "fr.js",
            "iso": "fr-FR",
            "name": "Français",
          },
        ],
        "seo": false,
        "vueI18n": Object {
          "fallbackLocale": "es",
          "silentTranslationWarn": false,
        },
      },
    ],
    Array [
      "@nuxtjs/pwa",
      Object {
        "manifest": false,
        "workbox": false,
      },
    ],
  ],
  "modulesDir": Array [
    "../../node_modules",
    "../../../node_modules",
    "../.tmp/simple/node_modules",
    "../.tmp/cli/node_modules",
    "../.tmp/advanced/node_modules",
    "../.tmp/spanish/node_modules",
  ],
  "plugins": Array [
    "~/plugins/axios",
    "~/plugins/vue-svgicon",
  ],
  "rootDir": "../..",
  "server": Object {
    "host": "127.0.0.1",
    "port": 3000,
  },
  "srcDir": "../../client",
  "statusfy": Object {
    "assets": Object {
      "mainLogo": "../.tmp/advanced/theme/default/img/logo.svg",
    },
    "dev": true,
    "iconSizes": Array [
      16,
      120,
      144,
      152,
      192,
      384,
      512,
    ],
    "langDir": "locales/",
    "locales": Array [
      Object {
        "code": "es",
        "iso": "es-ES",
        "name": "Español",
      },
      Object {
        "code": "fr",
        "iso": "fr-FR",
        "name": "Français",
      },
    ],
    "publicFilesPath": "../.tmp/spanish/public",
    "siteConfig": Object {
      "baseUrl": "https://demo-es.statusfy.co",
      "build": Object {
        "isStatic": false,
      },
      "content": Object {
        "dir": "content",
        "frontMatterFormat": "yaml",
        "systems": Array [
          "notificaciones",
          "monitoreo",
          "entrega-sitio",
        ],
      },
      "defaultLocale": "es",
      "description": "Descripción",
      "locales": Array [
        Object {
          "code": "es",
          "iso": "es-ES",
          "name": "Español",
        },
        Object {
          "code": "fr",
          "iso": "fr-FR",
          "name": "Français",
        },
      ],
      "manifest": true,
      "name": "Nombre Aplicación",
      "notifications": Object {
        "feeds": true,
        "icalendar": true,
        "support": false,
        "twitter": false,
      },
      "serviceWorker": true,
      "short_title": "Título",
      "sourceDir": "../.tmp/spanish",
      "theme": Object {
        "scheduled": Object {
          "position": "belowSystems",
        },
      },
      "title": "Título",
    },
    "sourceDir": "../.tmp/spanish",
    "version": "0.5.0",
  },
  "workbox": Object {
    "assetsURLPattern": "/static/(?!content).*$",
    "cacheId": "Nombre Aplicación",
    "cleanupOutdatedCaches": true,
    "globDirectory": "../.tmp/spanish/.statusfy/dist/client",
    "offlineAnalytics": true,
    "publicPath": "/static/",
    "runtimeCaching": Array [
      Object {
        "strategyOptions": Object {
          "cacheExpiration": Object {
            "maxAgeSeconds": 300,
            "maxEntries": 10,
          },
          "cacheName": "Nombre Aplicación_app_name_title_api",
        },
        "urlPattern": "(/static/content)?/api/v0/.*",
      },
    ],
  },
}
`;
